plugins/scaffolder-backend/src/scaffolder/tasks/TaskWorker.ts

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2021 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { PassThrough } from 'stream';
+import { Logger } from 'winston';
+import * as winston from 'winston';
+import { JsonValue } from '@backstage/config';
+import { TaskBroker, Task } from './types';
+import fs from 'fs-extra';
+import path from 'path';
+import { TemplateActionRegistry } from './TemplateConverter';
+
+type Options = {
+  logger: Logger;
+  taskBroker: TaskBroker;
+  workingDirectory: string;
+  actionRegistry: TemplateActionRegistry;
+};
+
+export class TaskWorker {
+  constructor(private readonly options: Options) {}
+
+  start() {
+    (async () => {
+      for (;;) {
+        const task = await this.options.taskBroker.claim();
+        await this.runOneTask(task);
```

Comment:
this will stop spinning at the first leaked exception - i see that runOneTask is wrapped in a large try/catch but the claim() makes me nervous as a reader of this code - maybe this whole `for` body could be wrapped in a try/catch to convey that things are handled no matter what